src/utils/convertPipes/index.js

```diff
@@ -20,68 +28,150 @@ const getAnswer = (ctx, answerId) =>
+  Date: (format, value) => transform(format, value),
+  DateRange: (format, value) => transform(format, value),
+  Number: (format, value) => transform(format, value)
 };
 
 const PIPE_TYPES = {
   answers: {
     retrieve: ({ id }, ctx) => getAnswer(ctx, id.toString()),
-    render: ({ id }) => `answers['answer${id}']`,
+    render: ({ id }) => `answer${id}`,
     getType: ({ type }) => type
   },
   metadata: {
     retrieve: ({ id }, ctx) => getMetadata(ctx, id.toString()),
-    render: ({ key }) => `metadata['${key}']`,
+    render: ({ key }) => `${key}`,
     getType: ({ type }) => type
   },
   variable: {
     render: () => `%(total)s`
   }
 };
 
-const convertElementToPipe = ($elem, ctx) => {
-  const { piped, ...elementData } = $elem.data();
+const parseHTML = html => {
+  return cheerio.load(html)("body");
+};
+
+const getPipedData = store => (element, ctx) => {
+  const { piped, ...elementData } = element.data();
   const pipeConfig = PIPE_TYPES[piped];
+
   if (piped === "variable") {
     return pipeConfig.render();
   }
+
   if (!pipeConfig) {
     return "";
   }
 
   const entity = pipeConfig.retrieve(elementData, ctx);
+
   if (!entity) {
     return "";
   }
+
   const output = pipeConfig.render(entity);
-  const dataType = pipeConfig.getType(entity);
+  const pipedType = pipeConfig.getType(entity);
+
+  const transformed = FILTER_MAP[pipedType];
+
+  const isText = transformed
+    ? `${transformed(pipedType, output).value}`
+    : `${output}`;
+
+  let placeholder = {};
+
+  if (transformed) {
+    let dateFormat;
+    if (entity.properties) {
+      dateFormat = entity.properties.format;
+    }
+
+    const { format, value, options } = transformed(pipedType, output);
+    placeholder = {
+      placeholder: removeDash(value),
+      transforms: [
+        {
+          transform: format,
+          arguments: options(
+            piped,
+            entity.key || `answer${entity.id}`,
+            dateFormat
+          )
+        }
+      ]
+    };
+  } else {
+    placeholder = {
+      placeholder: removeDash(isText),
+      value: {
+        source: piped,
+        identifier: entity.key || `answer${entity.id}`
+      }
+    };
+  }
 
-  const filter = FILTER_MAP[dataType];
+  store.placeholders = [...store.placeholders, placeholder];
 
-  return filter ? `{{ ${filter(output)} }}` : `{{ ${output} }}`;
-};
-
-const parseHTML = html => {
-  return cheerio.load(html)("body");
+  return `{${removeDash(isText)}}`;
 };
 
 const convertPipes = ctx => html => {
   if (!html) {
     return html;
   }
 
+  const store = {
+    text: "",
+    placeholders: []
+  };
+
   const $ = parseHTML(html);
 
-  $.find("[data-piped]").each((i, elem) => {
-    const $elem = cheerio(elem);
-    $elem.replaceWith(convertElementToPipe($elem, ctx));
+  $.find("[data-piped]").each((index, element) => {
+    const $elem = cheerio(element);
+    $elem.replaceWith(getPipedData(store)($elem, ctx));
   });
```

Comment:
Can we remove the `$` as a variable name please

Reply:
I think this is something to do with Cheerio. It's sort of like a mini jquery and that's the only reason I haven't touched it. 
https://github.com/cheeriojs/cheerio

Reply:
It's helpful due to the fact `$.find()` is the jQuery implementation of `find()` vs vanillaJS and they do two different things

Reply:
Ooooh. Truth be told, I didn't really look too far into why `$` was being used; I didn't realise it was server jQuery. That's actually pretty cool. Sorry, my bad. Cheers for explaining that.